cpp/tests/traversal/extract_bfs_paths_utils.cu

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph/graph.hpp>
+#include <cugraph/detail/utility_wrappers.hpp>
+
+#include <raft/handle.hpp>
+
+#include <rmm/device_uvector.hpp>
+
+#include <thrust/transform.h>
+
+template <bool multi_gpu, typename vertex_t>
+rmm::device_uvector<vertex_t> randomly_select_destinations(
+  raft::handle_t const& handle,
+  vertex_t number_of_vertices,
+  rmm::device_uvector<vertex_t> const& d_predecessors,
+  size_t num_paths_to_check,
+  uint64_t seed)
+{
```

Comment:
So, if I am not mistaken, this randomly selects ```num_paths_to_check``` vertices (that are not sources).

I think we can achieve this by

thrust::copy_if();
thrust::shuffle();

Reply:
And as this is only for test, another option is to do this in CPU, so no need to add this additional file and keep our codebase size in check.

Reply:
I can look into shuffle, that's one I haven't used before.

I can certainly look at a CPU implementation.  I was imagining that we might have other tests that might benefit from a function to randomly select vertices - although I started with it here because it was easier.

Reply:
Yeah... so, for personalized PageRank, we're randomly selecting personalization vertices as well (it's currently implemented as host code).

https://github.com/rapidsai/cugraph/blob/branch-21.10/cpp/tests/link_analysis/pagerank_test.cpp#L178
https://github.com/rapidsai/cugraph/blob/branch-21.10/cpp/tests/link_analysis/mg_pagerank_test.cpp#L104

We may better combine these and add something under tests/utilities?